scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/env python3.8
+
+import argparse
+import json
+import os
+import subprocess
+
+from typing import List, Union, Any
+from deepdiff import DeepDiff
+
+# {{{ JsonRpcProcess
+class MyEncoder(json.JSONEncoder):
+    """
+    Encodes an object in JSON
+    """
+    def default(self, o): # pylint: disable=E0202
+        return o.__dict__
```

Comment:
I haven't seen the rest of the file in detail yet but I have very serious doubts if this is a good idea. This basically ignores whether an object can be safely serialized and just dumps all of its fields and properties into JSON. This is going to silently break things one way or another.

Reply:
This part was taking from the pylspclient module to get me starting.